### PR TITLE
[Event Hubs] Logging improvements

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to enhance logging for some key scenarios. Changes include:

  - The addition of the MaxWaitTime and MaxBatchSize when reading events.

  - The addition of the end-to-end duration that it took to stop processing for a partition, including execution of the background continuation.

  - A new log that captures information for the partition processing cycle, including time taken to read events and process them.  This also contains a normalized start/end timestamp to allow for easier understanding of whether the host is unable to keep background tasks running without pauses.

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs [hunk 1]
```diff
@@ -2525,20 +2532,20 @@ public void UnexpectedException(string errorMessage)
         /// <param name="eventHubName">The name of the Event Hub that the processor is associated with.</param>
         /// <param name="partitionId">The identifier of the partition that the handler was invoked for.</param>
         /// <param name="operationId">An artificial identifier for the publishing operation.</param>
-        /// <param name="backoffSeconds">The number of seconds that the back-off will delay.</param>
-        /// <param name="backoffCount">The message for the exception that occurred.</param>
+        /// <param name="backOffSeconds">The number of seconds that the back-off will delay.</param>
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs:2535]: These are just capitalization corrections which got flagged by spell checking.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs [hunk 2]
```diff
@@ -961,7 +979,7 @@ async Task performProcessing()
             (
                 Task.Run(performProcessing),
                 partition,
-                readLastEnquedEventInformation,
+                readLastEnqueuedEventProperties,
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs:982]: This is just a typo fix to the name.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs [hunk 3]
```diff
@@ -3388,6 +3533,54 @@ public void UnexpectedException(string errorMessage)
...
+        ///   <see cref="EventSource.EventData"/> struct to avoid the parameter array allocation on the WriteEvent methods.
+        /// </summary>
+        ///
+        /// <param name="eventId">The identifier of the event.</param>
+        /// <param name="arg1">The first argument.</param>
+        /// <param name="arg2">The second argument.</param>
+        /// <param name="arg3">The third argument.</param>
+        /// <param name="arg4">The fourth argument.</param>
+        /// <param name="arg5">The fifth argument.</param>
+        ///
```
Comment (christothes) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs:3547]: fancy! Should we adopt this pattern in other perf critical paths such as Core?

**Review by christothes — APPROVED** [review]

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs (hunk 3 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs:3547]: Alex did a prototype for core once upon a time, and the micro-benchmarks didn't show much improvement, IIRC.    Struck me as odd, considering that without these you fall into a general `WriteEvent(params object[] args)` overload and pay for allocating the array.   The approach in the messaging libraries had a non-trivial positive impact.

https://github.com/Azure/azure-sdk-for-net/pull/33913

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs [hunk 4]
```diff
@@ -3388,6 +3533,54 @@ public void UnexpectedException(string errorMessage)
             }
         }
 
+        /// <summary>
+        ///   Writes an event with four string arguments and two value type arguments into a stack allocated
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs:3537]:
```suggestion
        ///   Writes an event with four string arguments and a value type argument into a stack allocated
```

**Pull request merged**